pipes/rules/assembly.rules

```diff
@@ -49,12 +50,29 @@ rule assemble_trinity:
         UGER        = config.get('UGER_queues', {}).get('short', '-l h_rt 04:00:00'),
         n_reads     = str(config["trinity_n_reads"]),
         logid       = "{sample}",
-        subsamp_bam = config["tmp_dir"]+'/'+config["subdirs"]["assembly"]+'/{sample}.subsamp.bam'
     run:
         makedirs(expand("{dir}/{subdir}",
             dir=[config["data_dir"],config["tmp_dir"]],
             subdir=config["subdirs"]["assembly"]))
-        shell("{config[bin_dir]}/assembly.py assemble_trinity {input.bam} {input.clipDb} {output.fasta} --n_reads={params.n_reads} --outReads {params.subsamp_bam} --threads {resources.threads} --JVMmemory 5g")
+        shell("{config[bin_dir]}/assembly.py assemble_trinity {input.bam} {input.clipDb} {output.fasta} --n_reads={params.n_reads} --outReads {output.subsamp_bam} --threads {resources.threads} --JVMmemory 5g")
+
+rule assemble_spades:
+    ''' This step runs the SPAdes assembler.
+    '''
+    input:  taxfilt_reads=config["data_dir"]+'/'+config["subdirs"]["per_sample"]+'/{sample}.taxfilt.bam',
+            clipDb = objectify_remote(config["trim_clip_db"]),
+            contigs_trinity=config["tmp_dir"] +'/'+config["subdirs"]["assembly"]+'/{sample}.assembly1-trinity.fasta'
```

Comment:
I like this new setup of the rules that allows the config.yaml to choose between two assembly pathways: trinity or trinity+spades. Are you sure you don't want to allow a spades-only option?

You could have two snakemake rules here: `assemble_spades` and `assemble_spades_from_trinity`. The latter would look mostly like your current rule (`assemble_spades`) except the output fasta filename would be .assembly1-trinity-spades.fasta. Then you could create a new rule `assemble_spades` that looks like this one but without the trinity contigs input and produces an output file called .assembly1-spades.fasta. Then add a comment to the config.yaml that the valid options for the assembler are either `trinity`, `spades`, or `trinity-spades`.

Reply:
I'm for adding these options, I was just wary of adding an extra rule, since I wasn't sure if the set of rules is part of the official public API?  My preference is that it isn't, but I wasn't sure what the consensus was.